controllers/nodehealthcheck_controller_test.go

```diff
@@ -1379,6 +1379,135 @@ var _ = Describe("Node Health Check CR", func() {
 			})
 		})
 	})
+
+	Context("Unhealthy condition checks", func() {
+
+		var (
+			r                   = &NodeHealthCheckReconciler{}
+			unhealthyConditions []v1alpha1.UnhealthyCondition
+			nodeConditions      []v1.NodeCondition
+
+			condType1         = v1.NodeConditionType("type1")
+			condType2         = v1.NodeConditionType("type2")
+			condStatusMatch   = v1.ConditionTrue
+			condStatusNoMatch = v1.ConditionUnknown
+
+			now                      = time.Now()
+			unhealthyDuration        = metav1.Duration{Duration: 10 * time.Second}
+			expireIn                 = 2 * time.Second
+			expiredTransitionTime    = metav1.Time{Time: now.Add(-unhealthyDuration.Duration).Add(-time.Second)}
+			notExpiredTransitionTime = metav1.Time{Time: now.Add(-unhealthyDuration.Duration).Add(expireIn)}
+
+			// this is always added in tested code
+			expireBuffer = time.Second
```

Comment:
IIUC this is to take into account [this `+ 1*time.Second` time buffer](https://github.com/medik8s/node-healthcheck-operator/blob/8e4b1384a7541f7740875097de32e3fd8d4fcb59/controllers/nodehealthcheck_controller.go#L431), isn't it?

Maybe it won't never change in the future, but have you considered using a `const` that can be shared with the test as well?

Reply:
Correct.

Usually I try to avoid sharing too many things between production code and test code. It increases the risk that undesired changes aren't detected by tests.

Reply:
good point :+1: